File: frontend-syscop/src/app/user.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  public url = 'http://localhost:8095/register-service/api/v1/register';



  constructor(private http: HttpClient) { }
  saveUser(user: any) {
    this.http.post(this.url, user, {responseType: 'text'} ).subscribe((data) => {
               console.log(data);
     });

    console.log(user);
  }

  // getUsers(): Observable<User[]> {
  //   return this
  //          .http
  //          .get<User[]>(`${this.url}/users`);
  // }
  // deleteuser(id) {
  //   return this.http.delete(`${this.url}/user/${id}`, {responseType: 'text'});

  // }
  // updateUser(): Observable<Object> {
  //   return this.http.put(`${this.url}/user`, User, {responseType: 'text'}) ;
  // }

}

File: userservice/src/main/java/com/stackroute/service/UserService.java
package com.stackroute.service;

import com.stackroute.model.User;
import com.stackroute.exception.UserAlreadyExistsException;
import com.stackroute.exception.UserNotFoundException;

import java.util.List;

public interface UserService {

        public User saveUser (User user) throws UserAlreadyExistsException;
        public List<User> getAllUsers();
        public User deleteUser(String userId) throws UserNotFoundException;
        public User updateUser(String userId, User user) throws UserNotFoundException;
    }

